tests/contract_testcases/solana/address_cast.sol

```diff
@@ -0,0 +1,8 @@
+contract c {
+	function foo() public {
+		// We have code that cast address type to ref address
+		// in fn sema::cast(). Ensure that this does not cause
+		// address values to be assignable.
```

Comment:
>  the test incorrectly assumed the value was not assignable.

This left me confused. If the test incorrectly assumed the value was not assignable, why are you testing here that the values should not be assignable?

Reply:
This PR has two commits. This is related to the first commit

Reply:
I don't really pay attention to commit messages because they are usually hidden in GitHub. Thanks for signaling that!